### DISCUSSION TITLE
Caching

### DISCUSSION THREAD

**Discussion by bkrew** [zero01101/openOutpaint#257] [Q&A] [answered]
I´m using this amazing tool regularly, my ssd with automatic1111 and all the tools slowely but surely are filling the disks. 
My question is, where does openoutpaint cache middlepictures or all the created you dont use ? 
Im not sure if this app clogs my system with cached images or raw data.

**Comment by zero01101** [comment]
so i could be entirely wrong, and i've only watched on windows so mac/linux could be very different, but while observing python exes with [procmon](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon), no temporary images were actually written to disk while using API calls - guessing they move directly from in-memory to a base64 representation?  while using webUI directly, images were observed writing to disk first via a [filename_parameters].tmp file in the defined output dir in webUI's settings, which immediately updated to a .png after the bytes were written out, but as far as openOutpaint is concerned, it's strictly in-memory until you use the save/upscale button :)

**Reply by bkrew** [reply]
Im not entirely sure, im quite new to github and programming per se. But that sounds reasonable. A soon as i started using Stable Diffusion, Easy LoRa and so on, i can watch my SSD´s get clogged up with temporary Data. I need to think about creating a pool where i store all the creations, Batch Data and so on... For me ist´s common to create like 20gb of waste in a few days 😂
I´m not sure if it´s worth to open a case, maybe you can answer me this, Openoutpaint worked fine for me for weeks now. But the last 2 weeks there is some odd behaviour. My api process gets killed during picture creation, you can see the canvas flash an the counter stops randomly. I have to restart everything to make it work again, the weird thing is that it just stops without error message.

**Comment by zero01101** [comment]
there should almost definitely be some kind of error message in the command prompt/terminal and/or in the f12 console in your web browser, unless you've installed some weird browser extension or something in that same time period that may be conflicting

**Comment by bkrew** [comment]
Hi, the error occured today. I copied some text from the console, maybe you can tell what the cause is. 
As soon the error occurs, openoutpaint is unresponsive, Even if the outpaint frame is moving. 
I have to close and reload the Web UI completely to make it work again, there are no further error messages.
Thanks in advance
Request URL:
http://127.0.0.1:7860/favicon.ico
Request Method:
GET
Status Code:
403 Forbidden
Remote Address:
127.0.0.1:7860
Referrer Policy:
strict-origin-when-cross-origin
Content-Length:
71
Content-Type:
application/json
Date:
Wed, 30 Aug 2023 12:28:41 GMT
Server:
uvicorn
X-Process-Time:
0.062
GET /favicon.ico HTTP/1.1
Accept: image/avif,image/webp,image/apng,image/svg+xml,image/*,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: de-DE,de;q=0.9,en-US;q=0.8,en;q=0.7
Connection: keep-alive
Cookie: _ga=GA1.1.1508341218.1692576391; X-CSRF-Token=b2facd9979a5d246ec5d62ac90f8c4fe3e8cdb776c154d8c2ba2e8e5c5ad4682; _ga_R1FN4KJKJH=GS1.1.1693354312.16.0.1693354312.0.0.0
Host: 127.0.0.1:7860
Referer: http://127.0.0.1:7860/
Sec-Fetch-Dest: image
Sec-Fetch-Mode: no-cors
Sec-Fetch-Site: same-origin
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36 OPR/101.0.0.0
sec-ch-ua: "Not/A)Brand";v="99", "Opera GX";v="101", "Chromium";v="115"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Windows"

**Comment by zero01101** [comment, marked as answer]
sorry for the delay; so that's gonna be unrelated unfortunately; that's just stating that the favicon (custom icon in the corner of the page tab) isn't being loaded, and wouldn't have anything to do with app functionality :/
